spec/unit_tests/moab/storage_services_spec.rb

```diff
@@ -3,6 +3,14 @@
 describe Moab::StorageServices do
   let(:eq_xml_opts) { { element_order: false, normalize_whitespace: true } }
 
+  it 'is a singleton' do
+    # see https://ruby-doc.org/stdlib-3.0.0/libdoc/singleton/rdoc/Singleton.html
+    described_class.instance
+    # providing an empty block returns a count, otherwise, an iterator is returned
+    class_instance_count = ObjectSpace.each_object(described_class) {} # rubocop:disable Lint/EmptyBlock
+    expect(class_instance_count).to eq 1
+  end
```

Comment:
this is the part i wanted to pair on:  this doesn't feel like a very realistic test, and so i'm not sure how worthwhile it is, but i had trouble writing a more realistic test that showed showed that this is a `Singleton`.

ideally, i'd like to test the desired behavior, which is that `@@repository` (now exposed by the `repository` method) is only initialized once.

Reply:
could you instantiate it and then do something else and confirm it's the same object with === ?

Reply:
the usage of the `StorageServices` classes is all via class methods, so i don't know that it really makes sense to instantiate them.  but i did end up adding tests that simulate calls from multiple threads, since that's the use case we're worried about, and those tests expect only one instantiation of `StorageRepository`.  hopefully that suffices?

Reply:
i think i may have a worthwhile test now in this PR?